bindings/python/tests/test_fn_sync.py

```diff
@@ -13,19 +19,33 @@ def test_list_devices():
     devices = list_devices()
     assert devices[0].name() == dev_name
 
+
 def test_get_device():
     dev_name = get_first_device_name("/dev/npu*")
-    dev_idx = int(dev_name.replace('npu', ''))
+    dev_idx = int(dev_name.replace("npu", ""))
     device = get_device(dev_idx)
     assert device.name() == dev_name
 
+
 def test_find_device_files():
     dev_name = get_first_device_name("/dev/npu*pe0-1")
     config = DeviceConfig(arch=Arch.Warboy, mode=DeviceMode.Fusion, count=1)
     devices = find_device_files(config)
     assert devices[0].filename() == dev_name
 
 
+def test_find_device_files_err():
+    dev_name = get_first_device_name("/dev/npu*pe0")
+    config = DeviceConfig.from_str(dev_name)
+    fd = os.open(f"/dev/{dev_name}", os.O_RDWR)
+    try:
+        _ = find_device_files(config)
+    except Exception as e:
+        assert str(e).endswith("found but still in use")
```

Comment:
This would pass even when `find_device_files` doesn't raise any exception. [`pytest.raises`](https://docs.pytest.org/en/7.4.x/reference/reference.html#pytest.raises) is a better choice:

```suggestion
    try:
        with pytest.raises(Exception, match=r"found but still in use$"):
            _ = find_device_files(config)
```

(Still has to keep `try` because of `os.close`.)

Reply:
reflect at [4d76a48](https://github.com/furiosa-ai/device-api/pull/94/commits/4d76a48ac395b4c162b8829e438093cfeaebef19)